src/decode/mod.rs

```diff
@@ -131,22 +131,24 @@ fn decode64<D: Decoder, P: Packer>(input: &[u8], decoder: D, packer: P) -> Resul
 
         if let Some(idx) = result.first_invalid {
             let idx = idx as usize;
-            if input[chunk_start + idx] == b'=' {
-                let rest_start = chunk_start + idx + 1;
-                let rest = &input[rest_start..];
-                let mut iter = rest
-                    .iter()
-                    .enumerate()
-                    .filter(|(_, c)| !c.is_ascii_whitespace());
-                trailer_length = match (iter.next(), iter.next()) {
-                    (None, _) => Some(1),
-                    (Some((_, b'=')), None) => Some(2),
-                    (Some((_, b'=')), Some((i, _))) | (Some((i, _)), _) => {
-                        return Err(Error::InvalidCharacter(rest_start + i))
-                    }
-                };
-            } else {
-                return Err(Error::InvalidCharacter(chunk_start + idx));
+            if chunk_start + idx < input.len() {
+                if input[chunk_start + idx] == b'=' {
+                    let rest_start = chunk_start + idx + 1;
+                    let rest = &input[rest_start..];
+                    let mut iter = rest
+                        .iter()
+                        .enumerate()
+                        .filter(|(_, c)| !c.is_ascii_whitespace());
+                    trailer_length = match (iter.next(), iter.next()) {
+                        (None, _) => Some(1),
+                        (Some((_, b'=')), None) => Some(2),
+                        (Some((_, b'=')), Some((i, _))) | (Some((i, _)), _) => {
+                            return Err(Error::InvalidCharacter(rest_start + i))
+                        }
+                    };
+                } else {
+                    return Err(Error::InvalidCharacter(chunk_start + idx));
+                }
```

Comment:
Please explain the need for this change

Reply:
Reverted - I thought this was an issue but it turns out not.